### PR TITLE
Add Python linter to CircleCI

### PR DESCRIPTION
**Pull request by mx-psi**
### What does this PR do?

- Add `flake8` linter with the `flake8-bugbear`plugin to CircleCI.
- Fix currently existing issues on Python files
- Add pre-commit hook for automatically running `flake8`
- Document pre-commit hook

### Motivation

A significant number of files have issues like missing, duplicate or unnecessary imports.
Adding `flake8` to the pipeline and the pre-commit hook will allow us to prevent many of these mistakes automatically.

### Additional Notes

- `flake8` configuration roughly follows the existing configuration at [integrations-core](https://datadoghq.dev/integrations-core/guidelines/style) but excluding all style-related options (which can be addressed with auto-formatting in a future PR).
- Linting can be disable per-line and per-file using the `noqa` directive (see examples in the changed code)

### Describe your test plan

- Tested the changed invoke tasks where it made sense
- Checked the CircleCI job output of the `python-linting` job

### CODE REVIEW

File: pkg/config/legacy/tests/config.py [hunk 1]
```diff
@@ -6,22 +6,16 @@
...
-from optparse import OptionParser, Values
 import os
-import platform
 import re
 from socket import gaierror, gethostbyname
 import string
 import sys
-import traceback
 from urlparse import urlparse
+from urllib.request import getproxies
```
Comment (mx-psi) [inline pkg/config/legacy/tests/config.py:18]: I am guessing this is the missing `getproxies` function from context, please double-check

File: test/e2e/containers/fake_datadog/app/api.py [hunk 2]
```diff
@@ -105,7 +105,7 @@ def get_series_from_query(q: dict):
     # group by
     # TODO
     last_open_brace, last_close_brace = query.rindex("{"), query.rindex("}")
-    group_by = query[last_open_brace + 1:last_close_brace].split(",")
+    group_by = query[last_open_brace + 1:last_close_brace].split(",")  # noqa: F841
```
Comment (mx-psi) [inline test/e2e/containers/fake_datadog/app/api.py:108]: Unused, deactivated linter because of the `TODO` comment.

File: tasks/release_tests.py [hunk 3]
```diff
@@ -25,7 +24,7 @@ def test_major_lower(self):
     def test_major_higher(self):
         version = self._get_random_version()
         increment = random.randint(1, 99)
-        self.assertTrue(release._is_version_higher(self._get_version(version["major"] + 1, version["minor"], version["patch"], version["rc"]),
+        self.assertTrue(release._is_version_higher(self._get_version(version["major"] + increment, version["minor"], version["patch"], version["rc"]),
```
Comment (mx-psi) [inline tasks/release_tests.py:27]: `increment` was unused, I am guessing the desired behavior was as changed based on the rest of tests.

**Review by apigirl — APPROVED** [review]
docs are fine

**Review by ahmed-mez — APPROVED** [review]
Changes in files owned by container integrations look good 👍

**Review by albertvaka — APPROVED** [review]
Good sanitization :)

**Review by hush-hush — APPROVED** [review]

**Pull request merged**